### PR TITLE
[WIP] Improve Get-ComputerInfo command

### PR DESCRIPTION
**Pull request by pougetat**
# PR Summary

The purpose of this PR is to improve the getcomputer-info command. Today the command retrieves all of the data from WMI and then filters the results based on the property names specified by the user.
This means that the following commands take the same amount of time to execute.

````powershell
Get-ComputerInfo
Get-ComputerInfo -Property BiosCharacteristics
````
In order to correct this I am storing the relationship between a WMI-queryable property and the WMI class to which it belongs in order to go from

```SQL
SELECT * FROM Win32_Bios
````
to 
````SQL
SELECT BiosCharacteristics FROM Win32_Bios
````

This PR is currently WIP since I have to add support for all of the properties a user can ask for but it sets the basic structure in place. I am hoping to get feedback on this general flow 😄 

@powercode also suggested removing the filtering capability and having the user simply combine Get-ComputerInfo with Select-Object to obtain feature parity but this can be had in a separate discussion (just thought it might be interesting to restate).

## PR Context

https://github.com/PowerShell/PowerShell/issues/9234

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Comment by vexx32** [comment]
@pougetat @rjmholt what's the status of this one?

See also #10349 -- that may be worth fixing in the scope of this PR if it is not already. 😄

File: src/Microsoft.PowerShell.Commands.Management/commands/management/CIMHelper.cs [hunk 1]
```diff
@@ -36,17 +37,90 @@ internal static class ClassNames
         internal const string MicrosoftNetworkAdapterNamespace = "root/StandardCimv2";
         internal const string DefaultQueryDialect = "WQL";
 
+        internal static Dictionary<string, CIMHelper.QueryInfo> QueryableProperties = new Dictionary<string, CIMHelper.QueryInfo>(){
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/CIMHelper.cs:40]: Maybe initialize this dictionary lazily?

**Review by daxian-dbw — COMMENTED** [review]
@pougetat Is this PR ready for review?

File: src/Microsoft.PowerShell.Commands.Management/commands/management/CIMHelper.cs [hunk 2]
```diff
@@ -36,17 +37,90 @@ internal static class ClassNames
...
         /// </returns>
-        internal static string WqlQueryAll(string from)
+        internal static string WqlQueryProperties(string from, List<string> requestedProperties)
         {
-            return "SELECT * from " + from;
+            var wmiPropertiesToQuery = requestedProperties
+            .Where(property => CIMHelper.QueryableProperties.ContainsKey(property) && CIMHelper.QueryableProperties[property].wmiClass == from)
+            .Select(property => CIMHelper.QueryableProperties[property].wmiPropertyName)
+            .ToList();
+            return wmiPropertiesToQuery.Count != 0 ? "SELECT " + string.Join(",", wmiPropertiesToQuery) + " from " + from : null;
```
Comment (rjmholt) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/CIMHelper.cs:123]:
```suggestion
    var wmiPropertiesToQuery = new List<string>();
    foreach (string propertyName in requestedProperties)
    {
        if (!CIMHelper.QueryableProperties.TryGetValue(propertyName, out CIMHelper.QueryInfo property
            || property.WmiClass != from)
        {
            continue;
        }
        
        wmiPropertiesToQuery.Add(property.WmiPropertyName);
    }
    
    if (wmiPropertiesToQuery.Count == 0)
    {
        return null;
    }
    
    return $"SELECT {string.Join(",", wmiPropertiesToQuery)} from {from}";
```

**Pull request closed without merging**